collector/perf_linux.go

```diff
@@ -35,37 +42,85 @@ func init() {
 // settings not all profiler values may be exposed on the target system at any
 // given time.
 type perfCollector struct {
-	perfHwProfilers    map[int]perf.HardwareProfiler
-	perfSwProfilers    map[int]perf.SoftwareProfiler
-	perfCacheProfilers map[int]perf.CacheProfiler
-	desc               map[string]*prometheus.Desc
+	hwProfilerCpuMap    map[*perf.HardwareProfiler]int
```

Comment:
I wouldn't expect to keep a map of pointers to interfaces. For example  [`NewSoftwareProfiler`](https://godoc.org/github.com/hodgesds/perf-utils#NewSoftwareProfiler) returns an interface that can be used directly.

Reply:
yea but I think you need them otherwise you can't generate a map of pointers to CPU ids.

Reply:
confirmed that this change (pointers to interfaces) works correctly, as the address pointed to can be used as a key for the CPU ID map. the code without this change does not work as mentioned here: https://github.com/prometheus/node_exporter/pull/1518#issuecomment-542880905

I think keeping this is correct -- I am using this in a lab setting and am getting the expected results now.